frontend/src/pages/application/DeviceGroups.vue

```diff
@@ -12,8 +12,11 @@
             <p>The device groups can then be set as the target in a DevOps Pipeline to update multiple devices in a single operation</p>
         </template>
     </SectionTopMenu>
-
-    <div v-if="deviceGroups?.length > 0" class="pt-4 space-y-6" data-el="pipelines-list">
+    <ff-loading
+        v-if="loading"
+        message="Loading Device Groups..."
+    />
+    <div v-else-if="deviceGroups?.length > 0" class="pt-4 space-y-6" data-el="pipelines-list">
```

Comment:
Instead of show/hide whole portions of the page, perhaps we should simply utelise the features of the data table like we do for "Loading Teams" and "Loading Users"?


https://flowfuse.github.io/forge-ui-components/#ff-data-table

> Property | Default | Description
> -- | -- | --
> `loading` | false | Shows a placeholder Loading message in place of the data.
> `loading-message` | 'Loading Data...' | If loading=true, then this message will be shown.
>